### PR TITLE
Allow separate MRIs per session (for longitudinal studies)

### PR DESCRIPTION
**Pull request by drammock**
WIP. Could use some advice about the best approach here; so far it *should* work (but yet untested!) for my use case but definitely won't work for the "normal" case that is currently supported on `main`. See the (small) diff for a prominent comment explaining the issue.

### Before merging …

- [ ] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: mne_bids_pipeline/steps/freesurfer/_01_recon_all.py [hunk 1]
```diff
@@ -77,21 +87,34 @@ def main(*, config) -> None:
...
     subjects_dir = Path(get_fs_subjects_dir(config))
     subjects_dir.mkdir(parents=True, exist_ok=True)
 
     with get_parallel_backend(config.exec_params):
         parallel, run_func = parallel_func(run_recon, exec_params=config.exec_params)
         parallel(
-            run_func(root_dir, subject, fs_bids_app, subjects_dir)
+            # TODO How can we infer if inner `for session in sessions` loop is needed?
+            #      Should we specify in `config`? As written, it will *only* work for
+            #      cases where subjects_dir has this layout:
```
Comment (larsoner) [inline mne_bids_pipeline/steps/freesurfer/_01_recon_all.py:106]: I think it's okay to use some filesystem inspection using whatever conditionals you need to construct the subj-session pairs, and then call the `parallel`. So something like
```
subj_session = list()
for subject in subjects:
    if mri_is_root_level(subject):  # TODO: Need to write this helper to do the filesystem inspection part!
        subj_session.append((subject, None))
    else:
        subj_session.extend((subject, session) for session in sessions)
parallel(run_func(...) for subject, session in subj_session)
```
where `session=None` would mean "root-level not session specific". Does that make sense? If not maybe I misunderstand the problem.

**Comment by drammock** [comment]
eh, this is not going to work as-is. There is an assumption baked-in in too many places that there's only a single folder level between `subjects_dir` and the various freesurfer files.  I'll explore making the folders something like `sub-116_ses-a/` rather than `sub-116/ses-a/`

**Comment by larsoner** [comment]
Makes sense

**Comment by drammock** [comment]
I think this is ready for review. Probably needs some addition to the docs so that folks other than me know how to actually make this work... I can add that along with whatever other changes y'all want.

**Comment by larsoner** [comment]
I think next you'd want to add `docs/source/v1.9.md.inc` to say that per-session MRIs are now supported.

Any idea if there is a dataset on openneuro that we could add to the test suite that would hit the new code path?

**Comment by drammock** [comment]
> I think next you'd want to add `docs/source/v1.9.md.inc` to say that per-session MRIs are now supported.

Done.  As for adding more thorough documentation elsewhere, I can't actually see where I would add it (since it doesn't involve a config value, and there's no dataset to add that could serve as an example).

> Any idea if there is a dataset on openneuro that we could add to the test suite that would hit the new code path?

I doubt it; OpenNeuro has zero datasets with modality=MEG and study_type=Longitudinal :(

**Comment by drammock** [comment]
Just checked the EEG Longitudinal studies too; none of them involve source reconstruction, only 1 even includes MRI at all, and it's only at 1 timepoint (https://openneuro.org/datasets/ds004024/versions/1.0.1)

**Comment by hoechenberger** [comment]
yes seems like it! do you want to take a look? just follow what i did in my previous bug reports :)

**Comment by larsoner** [comment]
https://github.com/OpenNeuroOrg/openneuro/issues/3145 and https://github.com/hoechenberger/openneuro-py/pull/187, openneuro folks usually resolve quickly so I say let's wait a day or two then restart. If they don't respond by then let's just merge.

**Comment by hoechenberger** [comment]
Issue has already been addressed upstream by OpenNeuro, restarting CI!

**Comment by hoechenberger** [comment]
@larsoner This is green. i won't have time to review, please move along if you're happy

**Comment by larsoner** [comment]
Thanks @drammock !

**Pull request merged**